src/Symfony/Component/DependencyInjection/Definition.php

```diff
@@ -341,17 +341,18 @@ public function setMethodCalls(array $calls = [])
      *
      * @param string $method    The method name to call
      * @param array  $arguments An array of arguments to pass to the method call
+     * @param bool   $useResult Whether the call returns the service instance or not
      *
      * @return $this
      *
      * @throws InvalidArgumentException on empty $method param
      */
-    public function addMethodCall($method, array $arguments = [])
+    public function addMethodCall($method, array $arguments = []/*, bool $useResult = false*/)
     {
         if (empty($method)) {
             throw new InvalidArgumentException('Method name cannot be empty.');
         }
-        $this->calls[] = [$method, $arguments];
+        $this->calls[] = 2 < \func_num_args() && \func_get_arg(2) ? [$method, $arguments, true] : [$method, $arguments];
```

Comment:
why not always defining the full array explicitly to make it easier for code consuming `getMethodCalls` while knowing it runs on Symfony 4.3+ (avoiding to hardcode the `?? false` everywhere) ?

Reply:
We did that before and that requires changing a lot of test cases, in the component and in bundles. That was a too high impact to me.